packages/web-components/src/components/va-file-input/va-file-input.tsx

```diff
@@ -0,0 +1,168 @@
+  styleUrl: 'va-file-input.css',
+  shadow: true,
+})
+
+export class VaFileInput {
+  @Element() el: HTMLElement;
+
+  /**
+   * The label for the button input.
+   */
+  @Prop() label?: string;
+
+  /**
+   * The name for the input element.
+   */
+  @Prop() name?: string;
+
+  /**
+   * The text displayed on the button
+   */
+  @Prop() buttonText: string;
+
+  /**
+   * Set the input to required and render the (Required) text.
+   */
+  @Prop() required?: boolean = false;
+
+  /**
+   * Allow the input to accept multiple files.
+   */
+  @Prop() multiple?: boolean = false;
+
+  /**
+   * A comma-separated list of unique file type specifiers.
+   */
+  @Prop() accept?: string;
+
+  /**
+   * The error message to render.
+   */
+  @Prop() error?: string;
+
+  /**
+   * Emit component-library-analytics events on the file input change event.
+   */
+   @Prop() enableAnalytics?: boolean = false;
+
+  /**
+   * The event emitted when the file input value changes.
+   */
+  @Event() vaChange: EventEmitter;
+
+  /**
+   * The event used to track usage of the component. This is emitted when the
+   * file input changes and enableAnalytics is true.
+   */
+   @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private handleChange = (e: Event) => {
+    const target = e.target as HTMLInputElement;
+    this.vaChange.emit({files: target.files});
+    /**
+     * Clear the original input, otherwise events will be triggered
+     * with empty file arrays and sometimes uploading a file twice will
+     * not work.
+     */
+    target.value = null;
+
+    if (this.enableAnalytics) {
+      this.componentLibraryAnalytics.emit({
+        componentName: 'va-file-input',
+        action: 'change',
+        details: {
+          label: this.label
+        },
+      });
+    }
+  };
+
+  private handleButtonClick = () => { 
+    this.el.shadowRoot.getElementById('fileInputField').click();
+  }
+
+  connectedCallback() {
+    i18next.on('languageChanged', () => {
+      forceUpdate(this.el);
+    });
+  }
+
+  disconnectedCallback() {
+    i18next.off('languageChanged');
+  }
+
+  render() {
+    const { 
+      label,
+      name, 
+      buttonText, 
+      required, 
+      multiple,
+      accept,
+      error,
+    } = this;
+    return (
+      <Host>
+        {label && (
+          <label htmlFor="fileInputButton">
+            {label}
+            {required && <span class="required">{i18next.t('required')}</span>}
+          </label>
+        )}
+        <slot></slot>
+        <span id="error-message" role="alert">
+          {error && (
+            <Fragment>
+              <span class="sr-only">{i18next.t('error')}</span> 
+              {error}
+            </Fragment>
+          )}
+        </span>
+        <va-button
+          id="fileInputButton"
+          aria-label={i18next.t(label)}
+          onClick={this.handleButtonClick}
```

Comment:
@bkjohnson I originally had a handler for `keypress` but we discovered it was firing both click and keypress events making the upload interface display twice. If focus is on the button though and space or enter are pressed, it will fire the click event (as a browser default behavior) so this should handle both mouse and keyboard events. Let me know if maybe you've discovered a bug with just having it like this though.

Reply:
Oh, that's nice. I didn't test it out, I just mentioned it because of [this recent issue](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/1096).